ipaserver/install/server/upgrade.py

```diff
@@ -1692,11 +1695,26 @@ def upgrade_configuration():
             )
         upgrade_pki(ca, fstore)
 
+        if kra.is_configured():
+            logger.info('[Ensuring ephemeralRequest is enabled in KRA]')
+            cainstance.backup_config(path=paths.KRA_CS_CFG_PATH)
+            value = installutils.get_directive(
+                paths.KRA_CS_CFG_PATH,
+                'kra.ephemeralRequests',
+                separator='=')
+            if value is None or value.lower() != 'true':
+                logger.info('Enabling ephemeralRequest')
+                kra.enable_ephemeral()
+            else:
+                logger.info('ephemeralRequest is already enabled')
+
     # several upgrade steps require running CA.  If CA is configured,
     # always run ca.start() because we need to wait until CA is really ready
     # by checking status using http
     if ca.is_configured():
         ca.start('pki-tomcat')
+    if kra.is_configured() and not kra.is_running():
+        kra.start('pki-tomcat')
```

Comment:
The CA and KRA (when configured) are both run under `pki-tomcatd@pki-tomcat.service` systemd service.  So the `ca.start('pki-tomcat')` will also start the KRA.  So I think this addition is redundant.

Reply:
I realize this, I did it for future-proofing in case the KRA service ever runs independently. I've left it in for now, I can remove it if you insist.

It should basically always fail.

Reply:
OK, I'm happy for it to be left in, but please mention the above-stated reason in the commit message.